external/java/src/main/java/ch/epfl/dedis/lib/omniledger/Instruction.java

```diff
@@ -186,13 +179,13 @@ public Request toDarcRequest() {
      * @param signers - the list of signers.
      * @throws CothorityCryptoException
      */
-    public void signBy(List<Signer> signers) throws CothorityCryptoException {
+    public void signBy(DarcId darcId, List<Signer> signers) throws CothorityCryptoException {
```

Comment:
And then the poor programmer who comes along later has to understand why there are two methods that seem to do the same thing, and choose one. This is a cultural difference btw Go and Java. As always I lean towards the YAGNI side of things (You Are Not Going To Need It, so don't write more code now than you know you need now.)

But if you reply and say, "no really, in Java we always give them 3 constructors to choose from, please write 2 more" I'm good.

I can make a vanilla one, a chocolate one, a chocolate/vanilla mixed with an optional AbstractSprinkleFactoryBuilder argument, etc. 🍨

Reply:
I like the AbstarctSprinkleFactoryBuilder, of course. If it could solve the climate catastrophy, too, that would be awesome!

For real (and not in this PR): we need to think how we can keep better track of the darc that is linked in an instance.